### PR TITLE
rgw: catches nobjects_begin() exceptions

### PR DESCRIPTION
**Pull request by max-licc**
The problem of nobjects_begin() crash occurred in the test environment. 

in RGWSI_RADOS::Pool::List::init() and log_list_init() , catch exceptions thrown by nobjects_begin().

Signed-off-by: lichaochao2_yewu@cmss.chinamobile.com

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
@max-licc thanks for the contribution! i've updated https://tracker.ceph.com/issues/54802 which tracks this crash

i see that the Signed-off-by check is failing. could you please update your commit messages to use the format it expects?
`Signed-off-by: Name <email>`

**Review by cbodley — APPROVED** [review]

File: src/rgw/driver/rados/rgw_rados.cc [hunk 1]
```diff
@@ -1422,10 +1422,21 @@ int RGWRados::log_list_init(const DoutPrefixProvider *dpp, const string& prefix,
...
-  *handle = (RGWAccessHandle)state;
-  return 0;
+  try {
+    state->prefix = prefix;
+    state->obit = state->io_ctx.nobjects_begin();
+    *handle = (RGWAccessHandle)state;
+    return 0;
+  } catch (const std::system_error& e) {
+    r = -e.code().value();
+    ldpp_dout(dpp, 10) << "nobjects_begin threw " << e.what()
```
Comment (mattbenjamin) [inline src/rgw/driver/rados/rgw_rados.cc:1432]: is 10 the right log level for this kind of error?

Comment (cbodley) [reply src/rgw/driver/rados/rgw_rados.cc:1432]: i think 10 is a good choice. if this ends up being a fatal error, we should already be logging that at level 0 or 1 further up the call stack

Comment (mattbenjamin) [reply src/rgw/driver/rados/rgw_rados.cc:1432]: will we have more than an error code at that point in the unwind?

**Comment by max-licc** [comment]
> @max-licc thanks for the contribution! i've updated https://tracker.ceph.com/issues/54802 which tracks this crash
> 
> i see that the Signed-off-by check is failing. could you please update your commit messages to use the format it expects? `Signed-off-by: Name <email>`

Thank you for your review. I have fixed it.

**Comment by ivancich** [comment]
Teuthology run:
* https://pulpito.ceph.com/ivancich-2023-02-17_00:14:49-rgw-wip-eric-testing-1-distro-default-smithi
* https://pulpito.ceph.com/ivancich-2023-02-17_00:15:25-rgw-wip-eric-testing-1-distro-default-smithi

**Comment by ivancich** [comment]
jenkins test make check

**Comment by ivancich** [comment]
This passed my QA run and I'm waiting for `make check` to complete so I can merge.

@max-licc : would you be so kind as to fill in the checklist above and determine whether any backports are needed and if so create an appropriate tracker? Thank you!

**Pull request merged**